Product Gallery Dialog: Refactor Interactivity API actions

danieldudzic opened the pull request:
This PR changes the Product Gallery Dialog Interactivity API action names and location.

<!-- Please do not remove any information from this pull request. Instead, add N/A or leave blank if not applicable -->

## What

Fixes #11311

## Why

Since the `openDialog` action is shared across components it doesn't make sense for it to live inside of the `Large Image`.

The `openDialog` and `closeDialog` names are a bit more informative about the actual actions performed by the action.

## Testing Instructions

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

_Please consider any edge cases this change may have, and also other areas of the product this may impact._


1. With a block theme, open `Appearance > Editor`.
2. Open `Templates > Single Product`.
3. Select `Convert to Blocks` if you are using the Classic Templates.
4. Add the `Product Gallery` block.
5. Ensure the Thumbnails are enabled.
6. Open a product page on the frontend **with multiple product images** (more than `Number of Thumbnails` setting) and test the frontend. Make sure the last thumbnail displays an overlay that opens the Dialog on click.
7. Make sure there are no regressions and the Dialog can still be triggered by clicking the Large Image.

* [x] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested (ie: it makes changes to tests, coding standards, docblocks, etc.). -->
* [x] Should be tested by the development team exclusively <!-- Check this box if this PR should be tested by the development team exclusively (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

## Screenshots or screencast

<!-- Any screenshots of UI changes will be helpful to include here. Leave blank if not applicable. -->

| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility

<!-- Check this documentation link (../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WooCommerce core, part of the feature plugin, or experimental. -->
Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [x] Experimental
* [ ] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:
* [x] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog
<!-- Provide a brief, descriptive summary of the changes in this PR. Include potential impacts on different parts of the product. Example: "Updated the checkout process to streamline the experience for users and reduce the number of steps." -->

> Add suggested changelog entry here.

danieldudzic committed:
Product Gallery Dialog: Change names of dialog actions and move them from the Large Image to the Product Gallery frontend file

danieldudzic committed:
Merge branch 'trunk' into fix/11311-product-gallery-refactor-handle-click-action

danieldudzic committed:
Merge branch 'trunk' into fix/11311-product-gallery-refactor-handle-click-action

danieldudzic committed:
Merge branch 'trunk' into fix/11311-product-gallery-refactor-handle-click-action

danieldudzic committed:
Merge branch 'trunk' into fix/11311-product-gallery-refactor-handle-click-action

roykho reviewed (changes requested):
So there is a regression here. I can no longer click on the gallery thumbnails to change the main image.

thealexandrelara commented on src/BlockTypes/ProductGalleryThumbnails.php:
```diff
@@ -157,7 +157,7 @@ protected function render( $attributes, $content, $block ) {
 								$processor->set_attribute(
 									'data-wc-on--click',
-									'actions.woocommerce.thumbnails.handleClick'
+									'actions.woocommerce.thumbnails.dialog.openDialog'
```
I believe that when changing the `handleClick` for the dialog, the editor also changed this that is related to the `thumbnails` by mistake and it causes the Thumbnails to not be clickable anymore.

thealexandrelara reviewed (changes requested):
Looks good! I added a comment regarding the `handleClick` function for the Thumbnails block that is currently not working.

danieldudzic committed:
Product Gallery: Update Interactivity API action names

danieldudzic commented:
@roykho @thealexandrelara Thank you for your reviews. I have now fixed the regression and updated other actions too, to follow the same semantic naming pattern. Please let me know if you'd change anything.

danieldudzic committed:
Merge branch 'trunk' into fix/11311-product-gallery-refactor-handle-click-action

danieldudzic committed:
Merge branch 'trunk' into fix/11311-product-gallery-refactor-handle-click-action

roykho reviewed (commented):
So I guess the only concern I have is the consistency. Because `dialog` has its own sub action but the others do not. So either we use sub actions for example `dialog` for all things housed inside the dialog component and for example `nextPrevious` for all things that belong there such as the `selectNextImage`...etc.

thealexandrelara commented:
> So I guess the only concern I have is the consistency. Because `dialog` has its own sub action but the others do not. So either we use sub actions for example `dialog` for all things housed inside the dialog component and for example `nextPrevious` for all things that belong there such as the `selectNextImage`...etc.

I agree with you. Since we now have more descriptive names (`openDialog` and `closeDialog`), maybe we don't need the `dialog` key anymore?

danieldudzic commented:
> I agree with you. Since we now have more descriptive names (`openDialog` and `closeDialog`), maybe we don't need the `dialog` key anymore?

That makes sense and would be my preference, too.

danieldudzic committed:
Merge branch 'trunk' into fix/11311-product-gallery-refactor-handle-click-action

danieldudzic committed:
Product Gallery Interactivity API actions: Remove the dialog key

danieldudzic committed:
Merge branch 'trunk' into fix/11311-product-gallery-refactor-handle-click-action

danieldudzic commented:
I'm closing this PR, as this is being already addressed in the migration to the new Interactivity API store: https://github.com/woocommerce/woocommerce-blocks/pull/11721
